rescheduler/rescheduler.go

```diff
@@ -291,50 +349,34 @@ func prepareNodeForPod(client *kube_client.Client, recorder kube_record.EventRec
 	return nil
 }
 
-func copyNode(node *kube_api.Node) (*kube_api.Node, error) {
-	objCopy, err := kube_api.Scheme.DeepCopy(node)
+func copyNode(node *apiv1.Node) (*apiv1.Node, error) {
+	objCopy, err := api.Scheme.DeepCopy(node)
 	if err != nil {
 		return nil, err
 	}
-	copied, ok := objCopy.(*kube_api.Node)
+	copied, ok := objCopy.(*apiv1.Node)
 	if !ok {
 		return nil, fmt.Errorf("expected Node, got %#v", objCopy)
 	}
 	return copied, nil
 }
 
-func addTaint(client *kube_client.Client, node *kube_api.Node, value string) error {
-	taints, err := kube_api.GetTaintsFromNodeAnnotations(node.Annotations)
-	if err != nil {
-		return err
-	}
-
-	taint := kube_api.Taint{
+func addTaint(client kube_client.Interface, node *apiv1.Node, value string) error {
+	node.Spec.Taints = append(node.Spec.Taints, apiv1.Taint{
```

Comment:
stealing a comment @liggitt made on another PR -- it's probably better to make sure the taint does not already exist, and only add it if it does not already exist.